### PR TITLE
Fix service session affinity e2e failure cases

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Fix service session affinity e2e failure cases - debuging...

**Which issue this PR fixes**:

xref #54571 #54524

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig network

### CODE REVIEW

File: test/e2e/framework/networking_utils.go
Comment (cmluciano) [inline test/e2e/framework/networking_utils.go]: typo 

Continer

File: test/e2e/framework/networking_utils.go
Comment (cmluciano) [inline test/e2e/framework/networking_utils.go]: typo 

Unexpected

**Review by cmluciano — CHANGES_REQUESTED** [review]
some typos

File: test/e2e/network/networking.go
Comment (spiffxp) [inline test/e2e/network/networking.go]: Would prefer to explain when/why, eg:

"Failed to get endpoints from test container, error: "

File: test/e2e/network/networking.go
Comment (spiffxp) [inline test/e2e/network/networking.go]: typo, Unexpected

File: test/e2e/framework/networking_utils.go
Comment (spiffxp) [inline test/e2e/framework/networking_utils.go]: Why the rename? Diff to me implies comparing or a delta between an old and new.  I don't think it's doing that here?  As a name, `GetEndpointsFromContainer` already tells me it's getting all of the container's endpoints?

File: test/e2e/network/networking.go
Comment (spiffxp) [inline test/e2e/network/networking.go]: Would prefer to explain when/why, eg:

"Failed to get endpoints from test container, error: "

File: test/e2e/network/networking.go
Comment (spiffxp) [inline test/e2e/network/networking.go]: typo, Unexpected

File: test/e2e/framework/networking_utils.go
Comment (spiffxp) [reply test/e2e/framework/networking_utils.go]: I guess it looks like the original impl of this was buggy and returned too soon? Still think the name fits the intent here

**Review by spiffxp — COMMENTED** [review]
a few nits

File: test/e2e/framework/networking_utils.go
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: I think this should be handled outside of `GetEndpointsFromContainer()`.

File: test/e2e/framework/networking_utils.go
Comment (MrHohn) [reply test/e2e/framework/networking_utils.go]: +1, `GetEndpointsFromContainer` seems to be a proper name.

File: test/e2e/framework/networking_utils.go
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: s/maxTries/tries

**Comment by m1093782566** [comment]
We can see 

[sig-network] Networking should function for client IP based session affinity: http
[sig-network] Networking should function for client IP based session affinity: udp

is passing in `pull-kubernetes-e2e-gce`

Should we trigger it more to see if it's still flaky?

**Comment by cmluciano** [comment]
/test pull-kubernetes-e2e-gce

triggering one more time

**Review by cmluciano — APPROVED** [review]

File: test/e2e/network/networking.go
Comment (MrHohn) [inline test/e2e/network/networking.go]: Remove debug logging.

File: test/e2e/network/networking.go
Comment (MrHohn) [inline test/e2e/network/networking.go]: Remove debug logging.

**Review by MrHohn — COMMENTED** [review]
/hold

We need to put the tests back into slow suite before merging.

File: test/e2e/framework/networking_utils.go
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: Humm, why shouldn't we return error here?

**Review by MrHohn — COMMENTED** [review]
/lgtm
/hold cancel

File: test/e2e/framework/networking_utils.go
Comment (MrHohn) [reply test/e2e/framework/networking_utils.go]: Never mind, this seems correct :)

**Pull request merged**
